### PR TITLE
Crash under WebAutomationSession::platformSimulateMouseInteraction when running pointerevents/pointerevent_click_during_capture.html?mouse-auxclick WPT

### PR DESCRIPTION
**Pull request by lilyspiniolas**
#### 07e1827a12519c7f6e8342941d9b8a62c26a7fe2
<pre>
Crash under WebAutomationSession::platformSimulateMouseInteraction when running pointerevents/pointerevent_click_during_capture.html?mouse-auxclick WPT
<a href="https://bugs.webkit.org/show_bug.cgi?id=280768">https://bugs.webkit.org/show_bug.cgi?id=280768</a>
<a href="https://rdar.apple.com/137068514">rdar://137068514</a>

Reviewed by Aditya Keerthi and Abrar Rahman Protyasha.

Abrar Protyasha determined that web-platform-test
&apos;pointerevent_click_during_capture.html?mouse-auxclick&apos; is crashing due to
dragEventType not getting updated when a right click is simulated in
WebAutomationSession::platformSimulateMouseInteraction. This causes an NSEvent
instantiation with an NSEventType of 0, resulting in a crash. Adding this missing
update fixes the test crash.

* Source/WebKit/UIProcess/Automation/mac/WebAutomationSessionMac.mm:
(WebKit::WebAutomationSession::platformSimulateMouseInteraction):

Canonical link: <a href="https://commits.webkit.org/284577@main">https://commits.webkit.org/284577@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/897940edbff9f583a7d1cc4e5ba1daa7e8c90e21

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/69884 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/49285 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/22637 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/73969 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/21042 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/57085 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/20893 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/55483 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/13950 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/72950 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/44903 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/60272 "Passed tests") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/35964 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/41568 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/19419 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/63499 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/18050 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/75684 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/14109 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/17290 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/63169 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/14144 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/60347 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/63101 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/15510 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/11115 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/4719 "Passed tests") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/45088 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/46162 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/47433 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/45903 "Built successfully") | | |

### CODE REVIEW

**Review by pxlcoder — APPROVED** [review]

**Review by aprotyas — APPROVED** [review]
The change LGTM but I'd suggest rewording parts of the commit message!

You want your commit title (and/or the bugzilla title) to reflect the user facing issue, and not the approach we took to fix it. Having said that, do you mind changing the commit title to something like:

```
Crash under WebAutomationSession::platformSimulateMouseInteraction when running pointerevents/pointerevent_click_during_capture.html?mouse-auxclick WPT
```

> is crashing due to dragEventType not getting updated when a right click

What is the reason for the crash? You should mention how we instantiate an NSEvent with a wrong `NSEventType` value (`0`).
